agent/pipeline_parser.go

```diff
@@ -32,6 +33,15 @@ func (p PipelineParser) Parse() (interface{}, error) {
 		errPrefix = fmt.Sprintf("Failed to parse %s", p.Filename)
 	}
 
+	// If interpolation is disabled, just parse and return
+	if p.NoInterpolation {
+		var result interface{}
```

Comment:
I think we may need to do the same logic as we do on line 50 (with the switch between a slice or a map?)

Reply:
We only do that so that we can `parseWithEnv` if it's a map. Without interpolation I don't think we need to do that, do we?

Reply:
Hmmmmm. We might end up with out-of-order env if we do that though.

Reply:
But we always had that, it only mattered once we started doing interpolation.